pkg/minikube/minikube.go

```diff
@@ -0,0 +1,183 @@
+}
+
+// checkMinikubeVersion validates that the user has the correct version of Minikube installed.
+// If not, it prompts the user to download a newer version before continuing.
+func checkMinikubeVersion() error {
+	versionCheck := exec.Command("minikube", "version", "--short")
+	out, err := versionCheck.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("minikube version: %w", err)
+	}
+	fmt.Printf("Minikube version is: %s\n", string(out))
+
+	userMinikubeVersion, err := parseMinikubeVersion(string(out))
+	if err != nil {
+		return fmt.Errorf("parsing minikube version: %w", err)
+	}
+	if userMinikubeVersion < minikubeVersion {
+		var resp string
+		fmt.Printf("WARNING: We require at least Minikube v%.2f, while you are using v%.2f\n", minikubeVersion, userMinikubeVersion)
+		fmt.Println("You can download a newer version from https://github.com/kubernetes/minikube/releases/")
+		fmt.Print("Continue anyway? (not recommended) [y/N]: ")
+		fmt.Scanf("%s", &resp)
+		if strings.ToLower(resp) != "y" {
+			fmt.Println("Installation stopped. Please upgrade minikube and run again")
+			os.Exit(0)
+		}
+	}
+
+	return nil
+}
+
+// checkForExistingCluster checks if the user already has a Minikube cluster. If so, it provides
+// the option of deleting the existing cluster and recreating it. If not, it proceeds to
+// creating a new cluster
+func checkForExistingCluster() error {
+
+	getClusters := exec.Command("minikube", "profile", "list")
+	out, err := getClusters.CombinedOutput()
+	if err != nil {
+		// there are no existing minikube profiles, the listing profiles command will error
+		// if there were no profiles, we simply want to create a new one and not stop the install
+		// so if the error is the "MK_USAGE_NO_PROFILE" error, we ignore it and continue onwards
+		if !strings.Contains(string(out), "MK_USAGE_NO_PROFILE") {
+			return fmt.Errorf("check cluster: %w", err)
+		}
+	}
+	// TODO Add tests for regex
+	r := regexp.MustCompile(clusterName)
+	matches := r.Match(out)
+	if matches {
+		var resp string
+		fmt.Print("Knative Cluster " + clusterName + " already installed.\nDelete and recreate [y/N]: ")
+		fmt.Scanf("%s", &resp)
+		if strings.ToLower(resp) != "y" {
+			fmt.Println("Installation skipped")
+			return nil
+		}
+		fmt.Println("deleting cluster...")
+		deleteCluster := exec.Command("minikube", "delete", "--profile", clusterName)
+		if err := deleteCluster.Run(); err != nil {
+			return fmt.Errorf("delete cluster: %w", err)
+		}
+		if err := createNewCluster(); err != nil {
+			return fmt.Errorf("new cluster: %w", err)
+		}
+	}
+
+	if err := createNewCluster(); err != nil {
+		return fmt.Errorf("new cluster: %w", err)
+	}
+
+	return nil
+}
+
+// createNewCluster creates a new Minikube cluster
+func createNewCluster() error {
+
+	fmt.Println("☸ Creating Minikube cluster...")
+
+	// create cluster and wait until ready
+	createCluster := exec.Command("minikube", "start", "--cpus", "3", "--profile", clusterName, "--wait", "all")
```

Comment:
@csantanapr @julz by default, minikube uses the Docker driver. Given that some of the motivation for offering minikube was for users who were having trouble with Docker, would it be better to use a VM driver here instead?

Something like

```golang
if runtime.GOOS == "darwin" {
    createCluster := exec.Command("minikube", "start", "--cpus", "3", "--vm", "hyperkit", "--profile", clusterName, "--wait", "all")
    ...
}

if runtime.GOOS == "linux" {
    createCluster := exec.Command("minikube", "start", "--cpus", "3", "--vm", "kvm2", "--profile", clusterName, "--wait", "all")
    ...
}

if runtime.GOOS == "windows" {
    createCluster := exec.Command("minikube", "start", "--cpus", "3", "--vm", "hyperv", "--profile", clusterName, "--wait", "all")
    ...
}
```

(would also update the plugin output to include a message about needing to install the proper VM / driver and a link to the docs on how to do so)

Not entirely sure this is warranted, but figured it was worth discussion.

Reply:
on Mac I tend to have docker stopped/off and minikube would use hyperkit.
I think a lot users would have docker setup on their mac/windows and I think is ok to use it if they have it but force them to have it.

Let minikube decide which one to use. User can use `minikube config set` to configure the driver to use.

Reply:
I think it's probably worth putting that info in the screen output, for example, something like this

```
$ kn quickstart minikube
Running Knative Quickstart using minikube
Minikube version is: v1.23.0

☸ Creating Minikube cluster...

Using the standard minikube driver by default...
If you wish to use a different driver, please configure minikube using 
    minikube config set --driver <your-driver>

....
```

Reply:
>I think it's probably worth putting that info in the screen output, for example, something like this

I think is a good idea to put that in the screen